### PR TITLE
added caltech101 dataset and example

### PR DESCRIPTION
**Pull request by marcuniq**
keras.datasets.caltech101.load_paths() downloads and extracts the Caltech101 dataset, splits it into train / test set and returns the paths to the images in the train / tests as well as the according labels. Clients have to crop / scale the images themselves; an implementation in examples/caltech101_cnn.py is provided.

### CODE REVIEW

File: examples/caltech101_cnn.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+import shutil
+import os
+from PIL import Image
+from resizeimage import resizeimage
+
+'''
+    Train a (fairly simple) deep CNN on the Caltech101 images dataset.
+    GPU run command:
+        THEANO_FLAGS=mode=FAST_RUN,device=gpu,floatX=float32 python caltech101_cnn.py
+'''
```
Comment (fchollet) [inline examples/caltech101_cnn.py]: Could you provide a few more details, as many people probably aren't familiar with Caltech101. What is this about, why is it interesting, what accuracy does your model reach and how long does it take to train?
